apps/emqx/src/emqx_config_handler.erl

```diff
@@ -232,9 +232,16 @@ process_update_request(ConfKeyPath, _Handlers, {remove, Opts}) ->
     BinKeyPath = bin_path(ConfKeyPath),
     case check_permissions(remove, BinKeyPath, OldRawConf, Opts) of
         allow ->
-            NewRawConf = emqx_map_lib:deep_remove(BinKeyPath, OldRawConf),
-            OverrideConf = remove_from_override_config(BinKeyPath, Opts),
-            {ok, NewRawConf, OverrideConf, Opts};
+            case emqx_map_lib:deep_find(BinKeyPath, OldRawConf) of
+                {ok, _} ->
+                    NewRawConf = emqx_map_lib:deep_remove(BinKeyPath, OldRawConf),
+                    OverrideConf = remove_from_override_config(BinKeyPath, Opts),
+                    {ok, NewRawConf, OverrideConf, Opts};
+                {not_found, _Path, _Map} ->
+                    {error,
+                        {not_found,
+                            <<"The congifuration object was not found and can therefore not be deleted">>}}
```

Comment:
Yes, that would be simpler but don't you think the exact same issue also exist for other config defined objects that can be deleted? Of course if such places also exit one could of course fix them in the same way (i.e., check if the object exist first and then delete it). If one do it in two steps like this, the change will be non-atomic so the possibility of successfully deleting the same item twice still exist but I guess this is a non-issue as the chance that this will happen in practice should be slim.

Reply:
Right, but I think only the HTTP APIs should differentiate their responses if a config is missing, rather than failing the whole config update (which could be triggered by sources other than HTTP API).  There's also a chance of a race condition between checking if a resource exists and deleting, but I don't think it's severe.  It'll simply return a success response to two or more clients that are trying to achieve the same goal.  :thinking: 

An alternative would be to call the removal as it is done today, and compare the config before and after to decide if anything was removed, since the config change is transactional.

Reply:
Fair enough, I have force pushed a new commit. Please, review again.

Btw, the issue was reported for e5.0.0-beta.5. Is it enough to merge to master or shall it be merged anywhere else?

Reply:
I think it's meant to target `master`, since it'll go for the next `e5.0.0`